File: Compilers-II/Mini-Asgn-1_CS18BTECH11042/q1.c
// Simple IO program to demonstrate how GCC compiler handles errors
// Shreyas Havaldar, CS18BTECH11042

#include<stdio.h>

int main(){

    printf("Shreyas\n");
    printf("Havaldar\n);
    printf("CSE\n");
    printf("18\n);



    return 0;
}

// On compiling the file, using gcc q1.c, all the errors in the file are displayed at once. All statements with errors are displayed together.
// Command: gcc q1.c
// Output:
// q1.c: In function ‘main’:
// q1.c:9:12: warning: missing terminating " character
//      printf("Havaldar\n);
//             ^
// q1.c:9:12: error: missing terminating " character
//      printf("Havaldar\n);
//             ^~~~~~~~~~~~~
// q1.c:10:20: error: expected ‘)’ before ‘;’ token
//      printf("CSE\n");
//                     ^
// q1.c:11:12: warning: missing terminating " character
//      printf("18\n);
//             ^
// q1.c:11:12: error: missing terminating " character
//      printf("18\n);
//             ^~~~~~~
// q1.c:10:5: warning: passing argument 1 of ‘printf’ makes pointer from integer without a cast [-Wint-conversion]
//      printf("CSE\n");
//      ^~~~~~
// In file included from q1.c:4:0:
// /usr/include/stdio.h:318:12: note: expected ‘const char * restrict’ but argument is of type ‘int’
//  extern int printf (const char *__restrict __format, ...);
//             ^~~~~~
// q1.c:16:1: warning: format not a string literal and no format arguments [-Wformat-security]
//  }
//  ^
// q1.c:16:1: error: expected declaration or statement at end of input

File: Compilers-II/Mini-Asgn-1_CS18BTECH11042/q1.py
# Simple IO program to demonstrate how Python interpreter handles errors
# Shreyas Havaldar, CS18BTECH11042

print("Shreyas\n")
print("Havaldar\n)
print("CSE\n")
print("18\n)

# On interpreting the file, the python interpreter scans the program line by line and as it finds error in line 5, it stops and immediately displays the error. It does not scan for further errors until this one is resolved.
# Command: python q1.py 
# Output:
#   File "q1.py", line 5
#     print("Havaldar\n)
#                      ^
# SyntaxError: EOL while scanning string literal